ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -232,6 +270,97 @@ extern "C" void *compileModule(string TraceName, Module *M,
   return ptr;
 }
 
+/// Write the string `S` in its entirety to the file descriptor `FD`.
+void writeString(int FD, string S) {
+  const char *Buf = S.c_str();
+
+  size_t Remain = strlen(Buf);
+  while (Remain) {
+    assert(Remain >= 0);
```

Comment:
IMHO this `assert` would better come before the `Remain -= ` line which could (if `write` is silly) underflow. `assert(Wrote <= Remain)` might be one idea?

Reply:
Please carefully check [fc913f5](https://github.com/ykjit/yk/pull/575/commits/fc913f52c738a6f558cd5330dc891a491542e314).

I realised that `write()` returns `ssize_t`.